src/core/libraries/np_trophy/trophy_ui.cpp

```diff
@@ -4,71 +4,86 @@
 #include <chrono>
 #include <imgui.h>
 #include "common/assert.h"
+#include "common/singleton.h"
 #include "imgui/imgui_std.h"
 #include "trophy_ui.h"
 
 using namespace ImGui;
-using namespace Libraries::NpTrophy;
+namespace Libraries::NpTrophy {
 
-TrophyUI::TrophyUI() {
+std::optional<TrophyUI> current_trophy_ui;
+std::queue<TrophyInfo> trophy_queue;
+
+TrophyUI::TrophyUI(std::filesystem::path trophyIconPath, std::string trophyName)
+                   : trophy_icon_path(trophyIconPath), trophy_name(trophyName) {
+    trophy_start_time = std::chrono::system_clock::now().time_since_epoch().count();
+    trophy_time_now = trophy_start_time;
+    if (std::filesystem::exists(trophy_icon_path)) {
+        trophy_icon = RefCountedTexture::DecodePngFile(trophy_icon_path);
+    } else {
+        LOG_ERROR(Lib_NpTrophy, "Couldnt load trophy icon at {}", trophy_icon_path.string());
+    }
     AddLayer(this);
 }
 
 TrophyUI::~TrophyUI() {
     Finish();
 }
 
-void Libraries::NpTrophy::TrophyUI::AddTrophyToQueue(int trophyId, std::string trophyName) {
-    TrophyInfo newInfo;
-    newInfo.trophyId = trophyId;
-    newInfo.trophyName = trophyName;
-    trophyQueue.push_back(newInfo);
-}
-
 void TrophyUI::Finish() {
     RemoveLayer(this);
 }
 
-bool displayingTrophy;
-std::chrono::steady_clock::time_point trophyStartedTime;
-
 void TrophyUI::Draw() {
     const auto& io = GetIO();
 
     const ImVec2 window_size{
-        std::min(io.DisplaySize.x, 200.f),
-        std::min(io.DisplaySize.y, 75.f),
+        std::min(io.DisplaySize.x, 250.f),
+        std::min(io.DisplaySize.y, 70.f),
     };
 
-    if (trophyQueue.size() != 0) {
-        if (!displayingTrophy) {
-            displayingTrophy = true;
-            trophyStartedTime = std::chrono::steady_clock::now();
-        }
-
-        std::chrono::steady_clock::time_point timeNow = std::chrono::steady_clock::now();
-        std::chrono::seconds duration =
-            std::chrono::duration_cast<std::chrono::seconds>(timeNow - trophyStartedTime);
+    SetNextWindowSize(window_size);
+    SetNextWindowCollapsed(false);
+    SetNextWindowPos(ImVec2(io.DisplaySize.x - 250, 50));
+    KeepNavHighlight();
 
-        if (duration.count() >= 5) {
-            trophyQueue.erase(trophyQueue.begin());
-            displayingTrophy = false;
+    if (Begin("Trophy Window", nullptr,
+              ImGuiWindowFlags_NoDecoration | ImGuiWindowFlags_NoSavedSettings |
+                  ImGuiWindowFlags_NoInputs)) {
+        if (trophy_icon) {
+            Image(trophy_icon.GetTexture().im_id, ImVec2(50, 50));
+            ImGui::SameLine();
+        } else {
+            // placeholder
+            ImGui::GetWindowDrawList()->AddRectFilled(ImVec2(1040, 60), ImVec2(1090, 110),
+                                                      GetColorU32(ImVec4{0.7f}));
```

Comment:
Instead of absolute position, it's better to use:
```suggestion
            const auto pos = GetCursorScreenPos();
            ImGui::GetWindowDrawList()->AddRectFilled(pos, pos + ImVec2{50.0f}, GetColorU32(ImVec4{0.7f}));
```